ee/codegen/src/generators/graph-attribute.ts

```diff
@@ -336,6 +336,36 @@ export class GraphAttribute extends AstNode {
       return;
     }
 
+    // Check if this is a cycle edge (target node already exists in the graph)
+    const isCycle = this.isNodeInBranch(targetNode, mutableAst);
+    const isSourceInGraph = this.isNodeInBranch(sourceNode, mutableAst);
+
+    if (
+      sourceNode &&
+      isCycle &&
+      isSourceInGraph &&
+      mutableAst.type === "set" &&
```

Comment:
[nit] We have a condition below for `mutableAst.type === "set"`. in fact, addEdgeToGraph is actually the union of addEdgeToEmpty, addEdgeToSource, etc for each mutableAst type. we've added those helpers for 2 of the 5 already, plan is to have for each. so would prefer if we nest this bit of logic within it.